.gitignore

```diff
@@ -12,8 +12,11 @@
 
 pubspec.lock
 
+# iOS and macOS dependencies
```

Comment:
In https://github.com/flutter/flutter/issues/148018 you mention you're going to update the plugin template, you could alternatively make the change in each respective .gitignores so when you migrate the example project it matches?  That's a lot more work though, up to you.

Reply:
> That's a lot more work though

FWIW, my understanding of why we have this root `.gitignore` is exactly that: that it was a pain to update every single package every time something got added to the gitignore upstream in the templates.

I don't love its existence, but since we already have it for this very scenario there's no real harm in expanding it to a new file. If I ever decide I dislike it enough to mass-update all the packages, it'll be the same amount of work regardless of the scope of the file.

Reply:
I don't feel strongly one way or another, I'm happy to do whatever folks think is the best practice here.

It does look like plugins' `.gitignore` files have diverged though. For example:

1. https://github.com/flutter/packages/blob/main/packages/ios_platform_images/.gitignore
2. https://github.com/flutter/packages/blob/main/packages/file_selector/file_selector_ios/.gitignore

Reply:
Also, not every plugin even has a `.gitignore` (ex. https://github.com/flutter/packages/tree/main/packages/video_player/video_player_avfoundation) . I would probably just do the root `.gitignore`